src/docs/truffle/getting-started/package-management-via-ethpm.md

```diff
@@ -1,45 +1,71 @@
+    - **must** conform to regex: `^[a-z][-a-z0-9]{0,255}$`
+2. A package version (ex. `1.0.0`)
+    - **should** conform to [semver](http://semver.org/)
+3. A manifest URI (ex. `ipfs`)
+    - any content-addressed uri where the [package manifest] is stored.
+    - IPFS is currently the only supported filestore in truffle
 
-You can also install a package at a specific version:
+An ethpm [manifest](https://ethpm.github.io/ethpm-spec/glossary.html#term-package-manifest) is simply a JSON object, that **must** comply with the [ethpm schema](https://github.com/ethpm/ethpm-spec/blob/master/spec/v3.spec.json). A manifest contains the various contract assets that can be included in an ethpm package; including bytecodes, source files, deployment data, compiler data, metadata and more.
 
-```shell
-$ truffle install <package name>@<version>
-```
+In ethpm, there is no central [registry](https://docs.ethpm.com/erc1319). Every project must deploy and manage their own package registry where they can publish their packages. However, Truffle ships with a non-permissioned, registry to which anybody is able to publish their packages.
 
-Like NPM, EthPM versions follow [semver](http://semver.org/). You can find a list of all available packages at [the Ethereum Package Registry](http://explorer.ethpm.com/).
+# ONLY INSTALL PACKAGES FROM TRUSTED REGISTRIES!
+Due to the nature of the blockchain, interacting with packaged code carries a much higher risk than in other code platforms (pypi, npm, etc). A single line of malicious code in an otherwise harmless package can be the difference between a succesful project or a disastrous hack.
 
-## Installing Dependencies
+Only install packages from registries that are controlled by accounts or organizations whom you trust. Just because a package is available, this does not establish that it is safe. Only install packages from registries controlled by trusted wallets. Validated registry addresses should be sourced directly from an organization / project's github repo or twitter account, and preferably connected to an ENS domain name.
 
-Your project can define an `ethpm.json` file that among other things can pin your project to specific dependencies and versions. To install all dependencies listed in the `ethpm.json` file, run:
+Truffle ships with a default unauthenticated registry, to make it easy to get started using ethpm. Practice extreme caution when interacting with the packages on this registry, as anybody can release a package on the default registry.
 
-```shell
-$ truffle install
-```
+## How to install an ethpm package
+
+### 1. Set your registry.
+By default, your Truffle project will be connected to the default Truffle registry. Some other registries to explore include...
+- `libraries.ethpm.eth`: Contains deployments of all [OpenZeppelin](https://github.com/OpenZeppelin/openzeppelin-contracts/) libraries across mainnet and all testnets available to use, so you never have to deploy a library again.
```

Comment:
Should we include the actual addresses here in parens?

Reply:
The addresses of the deployed libraries on each chain? All of them? Maybe just one for an example? Maybe including etherscan links could be useful? Why do you think it'd be useful to see the addresses here?

Reply:
Can they always use the ENS names or do they need to configure it? I was just thinking in case they don't want to deal with ENS. But if it comes with ENS out of the box then I suppose it doesn't matter. Maybe these are documented somewhere and can be linked to?

Reply:
To use ens names - you only need ENS to be enabled in your truffle-config. I've included a line linking to the relevant docs.